packages/nc-gui/components/smartsheet/toolbar/FilterInput.vue

```diff
@@ -0,0 +1,186 @@
+<script setup lang="ts">
+import type { ColumnType } from 'nocodb-sdk'
+import {
+  ColumnInj,
+  EditModeInj,
+  ReadonlyInj,
+  computed,
+  isBoolean,
+  isCurrency,
+  isDate,
+  isDateTime,
+  isDecimal,
+  isDuration,
+  isFloat,
+  isInt,
+  isMultiSelect,
+  isPercent,
+  isRating,
+  isSingleSelect,
+  isTextArea,
+  isTime,
+  isYear,
+  provide,
+  ref,
+  toRef,
+  useProject,
+} from '#imports'
+import type { Filter } from '~/lib'
+import SingleSelect from '~/components/cell/SingleSelect.vue'
+import MultiSelect from '~/components/cell/MultiSelect.vue'
+import DatePicker from '~/components/cell/DatePicker.vue'
+import YearPicker from '~/components/cell/YearPicker.vue'
+import DateTimePicker from '~/components/cell/DateTimePicker.vue'
+import TimePicker from '~/components/cell/TimePicker.vue'
+import Rating from '~/components/cell/Rating.vue'
+import Duration from '~/components/cell/Duration.vue'
+import Percent from '~/components/cell/Percent.vue'
+import Currency from '~/components/cell/Currency.vue'
+import Decimal from '~/components/cell/Decimal.vue'
+import Integer from '~/components/cell/Integer.vue'
+import Float from '~/components/cell/Float.vue'
+import Text from '~/components/cell/Text.vue'
+
+interface Props {
+  column: ColumnType
+  filter: Filter
+}
+
+interface Emits {
+  (event: 'updateFilterValue', model: any): void
+}
+
+const props = defineProps<Props>()
+const emit = defineEmits<Emits>()
+
+const column = toRef(props, 'column')
+const editEnabled = ref(true)
+
+provide(ColumnInj, column)
+provide(EditModeInj, readonly(editEnabled))
+provide(ReadonlyInj, ref(false))
+
+const checkTypeFunctions = {
+  isSingleSelect,
+  isMultiSelect,
+  isDate,
+  isYear,
+  isDateTime,
+  isTime,
+  isRating,
+  isDuration,
+  isPercent,
+  isCurrency,
+  isDecimal,
+  isInt,
+  isFloat,
+  isTextArea,
+}
+
+type FilterType = keyof typeof checkTypeFunctions
+
+const { sqlUi } = $(useProject())
+
+const abstractType = $computed(() => (column.value && sqlUi ? sqlUi.getAbstractType(column.value) : null))
```

Comment:
The root cause to the issue that I've posted is that `column.value.dt` is `null` for LTAR columns because they don't exist in the database and they are virtual. That's why in `getAbstractType(col)` (e.g. MysqlUI.ts in nocodb-sdk) would throw the error in the switch case because `col.dt` is `null` and `null` doesn't have  `toLowerCase()`.

https://github.com/nocodb/nocodb/blob/159908b103592040981af3b2f706f182f896f2c4/packages/nocodb-sdk/src/lib/sqlUi/MysqlUi.ts#L874-L875